base/shell/explorer/notifyiconscust.cpp

```diff
@@ -20,26 +20,194 @@
 
 #include "precomp.h"
 
+#define BITMAP_WIDTH 16
+#define BITMAP_HEIGHT 16
+
+#define IDC_NOTIFY_OPTIONS 5000
+
+typedef struct
+{
+    DWORD hwnd;
+    DWORD uId;
+    DWORD uCallbackMsg;
+    DWORD unkow[2];
+    HICON hIcon;
+    WCHAR text[64];
+}TRAYDATA;
```

Comment:
Where does this come from?

Reply:
ReactOS does not map the buffer at <tbbutton.dwData> in the same way as Windows. In any case, this is not the right way to approach the problem. I'm working on it.
It was my digression to find the handler for the icon. The only correct fields are (hIcone and text)

Reply:
Maybe ReactOS handles this different, but I was asking where this came from, not how reactos vs windows behaves.